api-call.py

```diff
@@ -1,29 +1,226 @@
+    """Prints data in a specified format
+
+    :param data: data fetched from API
+    """
+
+    text = json.dumps(data, indent=4)
     print(text)
 
 
+def show_tasks():
+    """Show tasks to choose as a Menu
+    """
+
+    config = load_from_config()
+    # Get the API options from the config
+    api_options = config['apiOptions']
+    print(config['generalStrings']['whichApi'])
+    for i, option in enumerate(api_options, start=0):
+        print(f"{i}. {option}")
+
+
+def get_user_choice():
+    """Get user input from menu
+
+    :returns: The user choice
+    """
+
+    config = load_from_config()
+    return input(config['generalStrings']['enterChoice'])
+
+
+def get_url(choice):
+    """Get URL for API
+
+    :param choice: user choice input
+    :returns: API URL Object after user choice
+    """
+
+    config = load_from_config()
+    access_token = get_access_token()
+    content_type = config['accessToken']['headers']['Content-Type']
+    header = {'Content-Type': content_type, 'Authorization': access_token}
+    url_obj = {}
+    if choice == '0':
+        url_obj = {'url': config['generalStrings']['exiting'],
+                   'parameters': {},
+                   'header': header}
+    elif choice == '1':
+        url_obj = {'url': config['apiUrls']['beaverBus'],
+                   'parameters': {},
+                   'header': header}
+    elif choice == '2':
+        url_obj = {'url': config['apiUrls']['terms'],
+                   'parameters': {},
+                   'header': header}
+    elif choice == '3':
+        user_date = input('Enter Date (yyyy-mm-dd): ')
+        url_obj = {'url_terms': config['apiUrls']['terms'],
+                   'url_textbooks': config['apiUrls']['textbooks'],
+                   'parameters': {'date': user_date},
+                   'header': header}
+    elif choice == '4':
+        route_id = input('Enter Route Number: ')
+        url_obj = {'url_routes': config['apiUrls']['routes'],
+                   'url_arrivals': config['apiUrls']['arrivals'],
+                   'url_vehicles': config['apiUrls']['vehicles'],
+                   'route_id': route_id,  'parameters': {},
+                   'header': header}
+    else:
+        print(config['generalStrings']['invalidChoice'])
+
+    return url_obj
+
+
+def get_text_books_with_term_date(url_obj):
+    """Get text books based on specified term and date
+    """
+
+    terms_data_response = get_api_data(url_obj['url_terms'],
+                                       url_obj['parameters'],
+                                       url_obj['header'])
+    academic_year = terms_data_response['data'][0]['attributes']['calendarYear']
+    term = terms_data_response['data'][0]['attributes']['season']
+    url_obj['textbooks_parameters'] = {'academicYear': academic_year,
+                                       'term': term,
+                                       'subject': 'CS',
+                                       'courseNumber': '161'}
```

Comment:
Should `subject` and `courseNumber` be fixed values? Or can a user search base on their use cases?

Reply:
It was just a test case because textbooks API isn't working publicly.